File: myproj26/main08-spread-operator.py
# name, *rest = ["Tom", 10, "Seoul"]

# print(name)
# print(rest)


numbers = [1, 2, 3]

new_numbers = [
    10,
    20,
    30,
    *numbers,
    40,
    50,
    60,
    *numbers,
    70,
    80,
    90,
    *numbers,
]

print(new_numbers)


tom = {
    "name": "Tom",
    "age": 10,
    "region": "Seoul",
}

# steve는 tom과 age/region이 같음
# tom을 참조하여 name만 변경해서
# 새로운 steve를 만든다

steve = dict(tom, name="Steve")

print(steve)

File: myproj26/main08-spread-operator.js

// const [name, ...rest] = ["Tom", 10, "Seoul"]

// console.log(name);
// console.log(rest);

const numbers = [1, 2, 3];

const new_numbers = [
    10, 20, 30,
    ...numbers,
    40, 50, 60,
    ...numbers,
    70, 80, 90,
    ...numbers,
];

console.log(new_numbers);

// --------------

const tom = {
    "name": "Tom",
    "age": 10,
    "region": "Seoul",
};

const steve = {
    ...tom,
    name: "Steve",
};

console.log(steve);
